### DISCUSSION TITLE
Some tips about git with PR | 使用 git 创建 PR 的一些建议

### DISCUSSION THREAD

**Discussion by zombieJ** [ant-design/ant-design#37051] [Q&A]
[English Version](#user-content-english) | [Chinese Version](#user-content-chinese)

# <a id="english">English Version</a>

If you don't know much about git branch management, here are some simple suggestions. It's more cumbersome, but always works:

## Create branch for PR

🔥🔥🔥 Be sure to create a corresponding new branch for each PR, rather than continuing development on your `master` and `feature` branches. 🔥🔥🔥

<img width="362" alt="截屏2022-08-15 上午11 36 04" src="https://user-images.githubusercontent.com/5378891/184572910-2762ce91-b6bf-4f3e-98c1-dc12331e59a7.png">

## How to sync my master with latest?

Directly click sync in github page: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork

## What if I accidentally publish a PR on `master`, and every subsequent PR always carries the previous commit?

Patch upstream (Skip if already done this):

```bash
git remote add upstream https://github.com/ant-design/ant-design.git
```

Sync remote upstream branch:

```bash
git checkout -t upstream/master tmp-master
```

Override brach:

```bash
git br -D master
git checkout -b master
git push -f
```

-----

# <a id="chinese">Chinese Version</a>

如果你对 git 分支管理不太了解，这里有一些简单的建议。它会比较麻烦，但是总是有效：

## 为 PR 创建分支

🔥🔥🔥 务必每次提交 PR 都要创建一个对应的新分支，而不是在你的 `master` 和 `feature` 分支上继续开发。 🔥🔥🔥

<img width="362" alt="截屏2022-08-15 上午11 36 04" src="https://user-images.githubusercontent.com/5378891/184572910-2762ce91-b6bf-4f3e-98c1-dc12331e59a7.png">

## 如何将我的 master 同步到最新

在 github 页面直接点击 sync 即可：https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork

## 如果我不小心在 `master` 上发布了一次 PR，之后每次发布 PR 总是会带上之前的 commit 该怎么办？

添加 upstream（如果添加过则不需要重复添加）：

```bash
git remote add upstream https://github.com/ant-design/ant-design.git
```

同步远程分支：

```bash
git checkout -t upstream/master tmp-master
```

覆盖分支：

```bash
git br -D master
git checkout -b master
git push -f
```

**Comment by afc163** [comment]
加到文档里？

**Reply by zombieJ** [reply]
嗯，攒完转正
